From bad69961dc46b116ff94a5dc98dc92fbf333e594 Mon Sep 17 00:00:00 2001
From: Ronak Shah <ronak.shah@xilinx.com>
Date: Mon, 30 Mar 2020 08:39:43 -0700
Subject: [PATCH 1306/1851] ASoC: xlnx: Fixed I2S sound card registration issue

commit fe7ce88e9518ae366391a05cade0897c735f4267 from
https://github.com/Xilinx/linux-xlnx.git

Fixed mapping issue between I2S machine/card driver and
I2S CPU driver for successful I2S sound card registration

Signed-off-by: Ronak Shah <ronak.shah@xilinx.com>
Tested-by: Daksha Rohit <dakshar@xilinx.com>
Reviewed-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 sound/soc/xilinx/xlnx_pl_snd_card.c | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/sound/soc/xilinx/xlnx_pl_snd_card.c b/sound/soc/xilinx/xlnx_pl_snd_card.c
index ab54827a677c..ddab8b3e733b 100644
--- a/sound/soc/xilinx/xlnx_pl_snd_card.c
+++ b/sound/soc/xilinx/xlnx_pl_snd_card.c
@@ -181,8 +181,13 @@ static const struct snd_soc_ops xlnx_spdif_card_ops = {
 	.hw_params = xlnx_spdif_card_hw_params,
 };
 
-SND_SOC_DAILINK_DEFS(xlnx_i2s,
-		     DAILINK_COMP_ARRAY(COMP_CPU("xilinx-i2s")),
+SND_SOC_DAILINK_DEFS(xlnx_i2s_capture,
+		     DAILINK_COMP_ARRAY(COMP_CPU("xlnx_i2s_capture")),
+		     DAILINK_COMP_ARRAY(COMP_DUMMY()),
+		     DAILINK_COMP_ARRAY(COMP_PLATFORM(NULL)));
+
+SND_SOC_DAILINK_DEFS(xlnx_i2s_playback,
+		     DAILINK_COMP_ARRAY(COMP_CPU("xlnx_i2s_playback")),
 		     DAILINK_COMP_ARRAY(COMP_DUMMY()),
 		     DAILINK_COMP_ARRAY(COMP_PLATFORM(NULL)));
 
@@ -215,12 +220,12 @@ static struct snd_soc_dai_link xlnx_snd_dai[][XLNX_MAX_PATHS] = {
 	[I2S_AUDIO] = {
 		{
 			.name = "xilinx-i2s_playback",
-			SND_SOC_DAILINK_REG(xlnx_i2s),
+			SND_SOC_DAILINK_REG(xlnx_i2s_playback),
 			.ops = &xlnx_i2s_card_ops,
 		},
 		{
 			.name = "xilinx-i2s_capture",
-			SND_SOC_DAILINK_REG(xlnx_i2s),
+			SND_SOC_DAILINK_REG(xlnx_i2s_capture),
 			.ops = &xlnx_i2s_card_ops,
 		},
 	},
-- 
2.31.1

